vllm/model_executor/layers/rejection_sampler.py

```diff
@@ -318,10 +328,18 @@ def _create_output(
         output.mul_(~after_false_mask).add_(
             recovered_token_ids.mul(after_false_mask))
 
+        # concat spec and nonspec token
+        output_with_bonus_tokens = torch.cat(
+            [output_with_bonus_tokens, non_spec_token_ids])
```

Comment:
`num_emitted_tokens` will include non-spec tokens, making the metric useless in capturing spec system efficiency (100% efficiency is defined every token accepted plus the bonus token)